### DISCUSSION TITLE
Some ideas from Coinbase style guide

### DISCUSSION THREAD

**Discussion by smol-ninja** [sablier-labs/evm-monorepo#906] [Ideas]
Thanks to @PaulRBerg, I came across [Coinbase style guide](https://github.com/coinbase/solidity-style-guide). I'd like to share some of ideas from it that that I liked.

1. [Event names should be past tense](https://github.com/coinbase/solidity-style-guide?tab=readme-ov-file#3-event-names-should-be-past-tense). For example, we use present tense for event names such as `Claim` and `Clawback` instead of `Claimed` and `Clawbacked`. Which ones do you like more?

2. [Avoid writing interfaces](https://github.com/coinbase/solidity-style-guide?tab=readme-ov-file#7-avoid-writing-interfaces). I don't fully agree with this because interfaces are useful but I agree with the point that it makes it harder to understand the code when NatSpecs are in the interfaces and not in the core contracts. So an idea could be to add NatSpecs to core contracts as well but how would that go? It will be repeated in interfaces and contracts.

3. [Newlines between tag types](https://github.com/coinbase/solidity-style-guide?tab=readme-ov-file#3-newlines-between-tag-types): this seems to be improving readability.

```solidity
/// @notice Creates a new MerkleLockup campaign with a LockupLinear distribution.
/// @dev Emits a {CreateMerkleLL} event.
/// @param baseParams Struct encapsulating the {SablierV2MerkleLockup} parameters, which are documented in
/// {DataTypes}.
/// @param lockupLinear The address of the {SablierV2LockupLinear} contract.
/// @param streamDurations The durations for each stream.
/// @param aggregateAmount The total amount of ERC-20 assets to be distributed to all recipients.
/// @param recipientCount The total number of recipients who are eligible to claim.
/// @return merkleLL The address of the newly created MerkleLockup contract.
```

becomes 

```solidity
/// @notice Creates a new MerkleLockup campaign with a LockupLinear distribution.
///
/// @dev Emits a {CreateMerkleLL} event.
///
/// @param baseParams Struct encapsulating the {SablierV2MerkleLockup} parameters, which are documented in
/// {DataTypes}.
/// @param lockupLinear The address of the {SablierV2LockupLinear} contract.
/// @param streamDurations The durations for each stream.
/// @param aggregateAmount The total amount of ERC-20 assets to be distributed to all recipients.
/// @param recipientCount The total number of recipients who are eligible to claim.
///
/// @return merkleLL The address of the newly created MerkleLockup contract.
```

@sablier-labs/solidity ??

**Comment by PaulRBerg** [comment]
Confirming reception, but I will review much later when we get to build OE and V2.3 (or whatever it will be called post-package tethering).

**Comment by PaulRBerg** [comment]
I review Coinbase's style guide now.

**Event names in past tense** - I see where they're coming from and I would have been opened to this design if we had a clean slate. But we don't have a clean slate. Refactoring all events would increase the maintenance cost in our subgraphs and other event indexing services. So I'm against this.

**Avoid interfaces** - against this. Interfaces help integrators and external users. The context switch cost for us and auditors is a small price to pay. Furthermore, deleting all interfaces would make our integrators' lives more difficult.

**Newlines between tag types** - sounds good. You are welcome to create an issue. Though this should be included in a future release (not V2.2).

**Reply by smol-ninja** [reply]
Thanks for taking the time to review this. I agree with all your points.

**Comment by smol-ninja** [comment]
Closing this discussion since there is no further call to action except the following:
- https://github.com/sablier-labs/v2-core/issues/953
